lib/loggingService/LogEntry.js

```diff
@@ -247,17 +247,11 @@ LogEntry.prototype.addDetails = function (args) {
 
 
 LogEntry.prototype.addData = function (data) {
-	data = serializeAny(data);
-
-	if (this.data) {
-		for (var key in data) {
-			if (data.hasOwnProperty(key)) {
-				this.data[key] = data[key];
-			}
-		}
-	} else {
-		this.data = data;
+	if (!this.data) {
+		this.data = [];
 	}
+
+	this.data = data.map((entry) => serializeAny(entry)).concat(this.data);
```

Comment:
hmmm this line works?
the arrow function isn't returning the value from serializeAny according to what is written here.
also you're overwriting this.data and forcing a new array to be constructed every time
why not simplify into the following?
```
this.data.push(...data.map(serializeAny))
```